File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/database/HttpHeaderValueCursorParser.java
package com.prateekj.snooper.networksnooper.database;

import android.database.Cursor;

import com.prateekj.snooper.database.CursorParser;
import com.prateekj.snooper.networksnooper.model.HttpHeaderValue;

import static android.provider.BaseColumns._ID;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_HEADER_VALUE;

public class HttpHeaderValueCursorParser implements CursorParser<HttpHeaderValue> {

  @Override
  public HttpHeaderValue parse(Cursor cursor) {
    HttpHeaderValue httpHeaderValue = new HttpHeaderValue();
    httpHeaderValue.setId(cursor.getInt(cursor.getColumnIndex(_ID)));
    httpHeaderValue.setValue(cursor.getString(cursor.getColumnIndex(COLUMN_HEADER_VALUE)));
    return httpHeaderValue;
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/database/HttpHeaderValueCursorParser.kt
package com.prateekj.snooper.networksnooper.database

import android.database.Cursor

import com.prateekj.snooper.database.CursorParser
import com.prateekj.snooper.networksnooper.model.HttpHeaderValue

import android.provider.BaseColumns._ID
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_HEADER_VALUE

class HttpHeaderValueCursorParser : CursorParser<HttpHeaderValue> {

  override fun parse(cursor: Cursor): HttpHeaderValue {
    return HttpHeaderValue().apply {
      setId(cursor.getInt(cursor.getColumnIndexOrThrow(_ID)))
      value = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_HEADER_VALUE))
    }
  }
}
